jupyterhub_moss/spawner.py

```diff
@@ -79,6 +78,12 @@ def _validate_partitions(self, proposal):
                 env.setdefault("add_to_path", True)
         return partitions
 
+    slurm_info_cmd = traitlets.Unicode(
```

Comment:
I'm wondering if this is flexible enough because the output is tightly coupled with the parsing done in `_get_slurm_info`.

Also, there is maybe a better/simpler way to retrieve this information from SLURM.

Reply:
I was thinking of something like:
```python
    slurm_info = traitlets.Callable(
        help="""Provides SLURM cluster information.

            It will be called with the Spawner as a single argument and return a mapping: partition name: SlurmInfo.
            It may be async.
    """,
    ).tag(config=True)
```
It leaves more freedom to refactor/extend the default `_get_slurm_info` and also more freedom when this needs to be tuned, but it is at the expense of more code when willing to override it.
Would this work in your case?

Reply:
There might be other ways to get that info from Slurm through its REST API, but that is not default. IMO, since batchspawner uses simple strings to define `batch_submit_cmd` (sbatch) and `batch_query_cmd` (squeue), I would stick to the same format for consistency. Even if somebody wants to use a REST call to get that information, it is very simple to do so through a command.

Reply:
I was thinking of still using `sinfo` from the command line, but to also allow defining the parsing of `sinfo` output because both are tightly coupled, but it is OK like that: this can be added later if needed.

Reply:
I see what you mean, tying up the `sinfo` command and the parser in a single option does make sense, but I would do that in a different PR.